pynmea2/types/proprietary/ash.py

```diff
@@ -0,0 +1,99 @@
+# -- ASHTECH -- #
+from decimal import Decimal
+# pylint: disable=wildcard-import,unused-wildcard-import
+from ... import nmea
+from ...nmea_utils import *
+""" Support for proprietary messages from Ashtech receivers.
+    Documentation: www.trimble.com/OEM_ReceiverHelp/v4.85/en/
+"""
+
+class ASH(nmea.ProprietarySentence):
+    sentence_types = {}
+    """
+        Generic Ashtech Response Message
+    """
+    def __new__(_cls, manufacturer, data):
+        '''
+            Return the correct sentence type based on the first field
+        '''
+        sentence_type = data[1]
+        name = manufacturer + 'R' + sentence_type
+        cls = _cls.sentence_types.get(name, _cls)
+        return super(ASH, cls).__new__(cls)
+
+    def __init__(self, manufacturer, data):
+        self.sentence_type = data[1]
+        super(ASH, self).__init__(manufacturer, data)
```

Comment:
You can avoid having to define the `Empty` and `Sentence Type` fields on all the concrete types below by passing `data[2:]` here instead